adapters/mobfoxpb/mobfoxpb.go

```diff
@@ -0,0 +1,129 @@
+package mobfoxpb
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/buger/jsonparser"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+// MobfoxpbAdapter adapter structure
+type MobfoxpbAdapter struct {
+	URI string
+}
+
+// NewMobfoxpbBidder Initializes the Bidder
+func NewMobfoxpbBidder(endpoint string) *MobfoxpbAdapter {
+	return &MobfoxpbAdapter{
+		URI: endpoint,
+	}
+}
+
+// MakeRequests create bid request for mobfoxpb demand
+func (a *MobfoxpbAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+	var err error
+	var tagID string
+
+	var adapterRequests []*adapters.RequestData
+
+	reqCopy := *request
+	for _, imp := range request.Imp {
+		tagID, err = jsonparser.GetString(imp.Ext, "bidder", "TagID")
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		imp.TagID = tagID
+		reqCopy.Imp = []openrtb.Imp{imp}
+		adapterReq, err := a.makeRequest(&reqCopy)
+		if err != nil {
+			errs = append(errs, err)
+		}
+		if adapterReq != nil {
+			adapterRequests = append(adapterRequests, adapterReq)
+			break
```

Comment:
Why `break` only after the first imp? Are you sure you want to discard all other imps in the request if there are more than one? If you're bidder only supports 1 imp per request, you can create multiple `adapterRequests` for each imp and they will all be sent as separate requests to your bidder

Reply:
For now , there wouldn't be multiple imressions. We will work on multy bidding in next iteration

Reply:
I think that's fine. Please be clear in your docs that only the first imp is used and the others are ignored. I'd also like to see removal of the for loop since you're only looking at the first one. Something like:

```
imp := request.Imp[0]

tagID, err = jsonparser.GetString(imp.Ext, "bidder", "TagID")
if err != nil {
        return nil, []error{&errortypes.BadInput{
            Message: fmt.Sprintf("unable to read imp[0].ext.bidder.TagID (%v)", err)}}
}

reqCopy := *request
reqCopy.Imp = []openrtb.Imp{imp[0]}

adapterReq, err := a.makeRequest(&reqCopy)
if err != nil {
    return nil, []error{err}
}

return []*adapters.RequestData{adapterRequest}, nil
```

Reply:
I've made changes, which docs do you mean? prebid.github.io dev-docs? maybe there is some formal parameter which shows that adapter excepts multi imp or single one ?